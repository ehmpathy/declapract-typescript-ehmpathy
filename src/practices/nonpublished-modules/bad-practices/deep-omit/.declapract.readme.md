was replaced with `domain-objects#omitAutogeneratedValues`
