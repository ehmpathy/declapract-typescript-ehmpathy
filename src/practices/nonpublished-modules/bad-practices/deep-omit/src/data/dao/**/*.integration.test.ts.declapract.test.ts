import { fix } from './*.integration.test.ts.declapract';

describe('*.integration.test.ts', () => {
  it('should replace deep omit with omitAutogeneratedValues correctly', async () => {
    const exampleBadPracticeContents = `
import {
  DatabaseConnection,
  getDatabaseConnection,
} from '../../../utils/database/getDatabaseConnection';
import { findById } from './findById.query';
import { upsert } from './upsert.query';
import { deepOmit } from '../../../__nonpublished_modules__/deep-omit';
import { createExampleEmail } from '../../../__test_utils__/createExampleEmail';

describe('findById', () => {
  let dbConnection: DatabaseConnection;
  beforeAll(async () => {
    dbConnection = await getDatabaseConnection();
  });
  afterAll(async () => {
    await dbConnection.end();
  });
  it('should be able to find an upserted record', async () => {
    const upserted = await upsert({
      dbConnection,
      email: createExampleEmail(),
    });
    const found = await findById({ dbConnection, id: upserted.id! });
    expect(deepOmit(upserted, ['id'])).toEqual(deepOmit(found!, ['id']));
  });
});
    `.trim();
    const { contents: fixedContents } = await fix(
      exampleBadPracticeContents,
      {} as any,
    );
    console.log(fixedContents);
    expect(fixedContents).toContain(
      'expect(omitAutogeneratedValues(upserted)).toEqual(omitAutogeneratedValues(found!))',
    );
    expect(fixedContents).toMatchSnapshot();
  });
});
