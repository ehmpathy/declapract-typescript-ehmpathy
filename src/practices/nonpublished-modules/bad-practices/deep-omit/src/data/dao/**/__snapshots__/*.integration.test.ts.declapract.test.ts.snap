// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`*.integration.test.ts should replace deep omit with omitAutogeneratedValues correctly 1`] = `
"import {
  DatabaseConnection,
  getDatabaseConnection,
} from '../../../utils/database/getDatabaseConnection';
import { findById } from './findById.query';
import { upsert } from './upsert.query';
import { omitAutogeneratedValues } from 'domain-objects';
import { createExampleEmail } from '../../../__test_utils__/createExampleEmail';

describe('findById', () => {
  let dbConnection: DatabaseConnection;
  beforeAll(async () => {
    dbConnection = await getDatabaseConnection();
  });
  afterAll(async () => {
    await dbConnection.end();
  });
  it('should be able to find an upserted record', async () => {
    const upserted = await upsert({
      dbConnection,
      email: createExampleEmail(),
    });
    const found = await findById({ dbConnection, id: upserted.id! });
    expect(omitAutogeneratedValues(upserted)).toEqual(omitAutogeneratedValues(found!));
  });
});"
`;
